presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1173,8 +1173,8 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
     {
         verifyJvmTimeZone();
 
-        if (getExternalLocation(tableMetadata.getProperties()) != null) {
-            throw new PrestoException(NOT_SUPPORTED, "External tables cannot be created using CREATE TABLE AS");
+        if (!writesToNonManagedTablesEnabled && getExternalLocation(tableMetadata.getProperties()) != null) {
```

Comment:
One more thing, I think here we should check both: `writesToNonManagedTablesEnabled` and `createsOfNonManagedTablesEnabled`

Reply:
You're right, but any idea why that case isn't being caught by https://github.com/prestosql/presto/blob/master/presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java#L789

Reply:
AFAIR that place is only for create without data (one-off create)
and this code is only for create with data (prepare-write-commit)